### PR TITLE
Updated tokengen

### PR DESCRIPTION
**Pull request by vdbaan**
- Instead of removing all cookies, removed only the cookie that you want to target

### CODE REVIEW

**Comment by psiinon** [comment]
@vdbaan - did you see the build failure?

**Comment by vdbaan** [comment]
@psiinon - the number fuzzer should not be part of this pull request. I don't know why that joined. Will fix it today though.

File: src/org/zaproxy/zap/extension/fuzz/payloads/generator/NumberPayloadGenerator.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/generator/NumberPayloadGenerator.java]: Should be the ZAP license header.

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: Should be the ZAP license header.

File: src/org/zaproxy/zap/extension/fuzz/payloads/generator/NumberPayloadGenerator.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    public ResettableAutoCloseableIterator<DefaultPayload> iterator() {
+        return this;
+    }
+
+    @Override
+    public PayloadGenerator<DefaultPayload> copy() {
+        return this;
+    }
+
+    long pos;
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/generator/NumberPayloadGenerator.java]: Should probably be moved up to line 19/20

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+/**
+ *
+ * @author steven
+ */
+public class NumberPayloadGeneratorAdapterUIHandler implements
+        PayloadGeneratorUIHandler<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUI> {
+
+    private static final Logger LOGGER = Logger.getLogger(NumberPayloadGeneratorAdapterUIHandler.class);
+//    private static final String PAYLOAD_GENERATOR_NAME = Constant.messages.getString("fuzz.payloads.generator.number.name");
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: Should be removed

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    private static final String PAYLOAD_GENERATOR_NAME = "Numberzz";
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: Should use an i18n string (similar to what was commented out above).

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+        }
+
+        @Override
+        public String getName() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public String getDescription() {
+            return PAYLOAD_GENERATOR_NAME;
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: While I kind of understand that a number generator should be somewhat self describing, it's probably worth being slightly more specific here. Consider `PAYLOAD_GENERATOR_DESC` with another i18n string.

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+
+            GroupLayout layout = new GroupLayout(fieldsPanel);
+            fieldsPanel.setLayout(layout);
+            layout.setAutoCreateGaps(true);
+
+            fromField = new JTextField();
+            toField = new JTextField();
+            stepField = new JTextField();
+
+            JLabel fromLabel = new JLabel("From:");
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: This label should use an i18n string.

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+            fromLabel.setLabelFor(fromField);
+
+            JLabel toLabel = new JLabel("To:");
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: This label should use an i18n string.

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+            toLabel.setLabelFor(toField);
+
+            JLabel stepLabel = new JLabel("Step:");
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: This label should use an i18n string. (Should this be "Increment" for consistency?)

File: src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+        public JPanel getComponent() {
+            return fieldsPanel;
+        }
+
+        @Override
+        public void setPayloadGeneratorUI(NumberPayloadGeneratorUI payloadGeneratorUI) {
+            oldGenerator = payloadGeneratorUI;
+            fromField.setText(""+payloadGeneratorUI.getPayloadGenerator().getFrom());
+            toField.setText(""+payloadGeneratorUI.getPayloadGenerator().getTo());
+            stepField.setText(""+payloadGeneratorUI.getPayloadGenerator().getStep());
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java]: Is the empty string concatenation really necessary here?

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 11]
```diff
@@ -117,6 +117,16 @@ public Object getValueAt(int rowIndex, int columnIndex) {
         }
     }
 
+    // SB: Make copying of info easier
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: This comment probably isn't necessary.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 12]
```diff
@@ -117,6 +117,16 @@ public Object getValueAt(int rowIndex, int columnIndex) {
...
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+            if(getColumnClass(columnIndex) == String.class) {
+                return true;
+            }
+
+        return super.isCellEditable(rowIndex, columnIndex); //To change body of generated methods, choose Tools | Templates.
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: This comment should be removed.

**Comment by kingthorin** [comment]
@vdbaan this seems to contain contributions on two different items.

Could you split off the fuzz contributions and submit them separately?

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 13]
```diff
@@ -62,8 +77,11 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				// SB: Only remove the cookie that you want to analyse
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: The removal was actually done above.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 14]
```diff
@@ -62,8 +77,11 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				// SB: Only remove the cookie that you want to analyse
+				// msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: No longer needed?

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 15]
```diff
@@ -62,8 +77,11 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				// SB: Only remove the cookie that you want to analyse
+				// msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				msg.getRequestHeader().setCookieParams(cookies);
 				this.getHttpSender().sendAndReceive(msg, true);
+				Thread.sleep (10); // Not to flood all the requests
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Is this really necessary?

If there's a timing issue (un-related to the cookie changes) then this should be a separate issue/pull.

**Comment by kingthorin** [comment]
@vdbaan this seems to contain contributions on two different items.

Could you split off the fuzz contributions and submit them separately?

**Comment by vdbaan** [comment]
@kingthorin I'll retract this pull request and submit two separate ones which also will have your comments addressed.
Thank you for your help.

**Comment by kingthorin** [comment]
@vdbaan thanks! :smiley:

**Comment by thc202** [comment]
@vdbaan does it worth adding more review comments to this pull request? Or, better wait for the new ones?

P.S. You could still use this pull request for tokengen changes.

**Comment by vdbaan** [comment]
@thc202 add them  I still have to split the request it and will try to incorporate all comments that I receive.

**Comment by thc202** [comment]
OK, will do.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 14 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 13 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 12 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: Removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 11 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: Removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 15 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Removed.
I'll add a separate pull for this as it should be included, there only is a timing issue when you use this generator against an locally installed app.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 16]
```diff
@@ -1,10 +1,10 @@
 /*
  * Zed Attack Proxy (ZAP) and its related class files.
- * 
+ *
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: This type of indentation/formatting changes are better if done in another commit, if you really want to do them. [1]

[1] https://github.com/zaproxy/zaproxy/wiki/DevGuidelines

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 17]
```diff
@@ -117,6 +117,15 @@ public Object getValueAt(int rowIndex, int columnIndex) {
         }
     }
 
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: How about allow cell selection instead? One could use the usual keyboard shortcut to copy individual cell values (it also works with non-String ones).

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 18]
```diff
@@ -117,6 +117,15 @@ public Object getValueAt(int rowIndex, int columnIndex) {
...
+            if(getColumnClass(columnIndex) == String.class) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: This code block should have one less indentation level.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 19]
```diff
@@ -117,6 +117,15 @@ public Object getValueAt(int rowIndex, int columnIndex) {
...
+                return true;
+            }
+
+        return super.isCellEditable(rowIndex, columnIndex);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: This could just return false.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 20]
```diff
@@ -1,33 +1,37 @@
 /*
  * Zed Attack Proxy (ZAP) and its related class files.
- * 
+ *
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Same comment regarding indentation/formatting changes.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 21]
```diff
@@ -1,33 +1,37 @@
...
 package org.zaproxy.zap.extension.tokengen;
 
+import java.util.Iterator;
+import java.util.TreeSet;
+
 import javax.swing.SwingWorker;
 
 import org.apache.log4j.Logger;
 import org.parosproxy.paros.model.Model;
 import org.parosproxy.paros.network.HttpHeader;
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: This import is no longer in use.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 22]
```diff
@@ -47,6 +51,17 @@ private HttpSender getHttpSender() {
 
 	@Override
 	protected Void doInBackground() throws Exception {
+		 TreeSet<HtmlParameter> cookies = this.httpMessage.getCookieParams();
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: It should get only the cookies of the request header.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 23]
```diff
@@ -47,6 +51,17 @@ private HttpSender getHttpSender() {
 
 	@Override
 	protected Void doInBackground() throws Exception {
+		 TreeSet<HtmlParameter> cookies = this.httpMessage.getCookieParams();
+		 if (targetToken.getType() == HtmlParameter.Type.cookie) {
+			 Iterator<HtmlParameter> iter = cookies.iterator();
+			 while (iter.hasNext()) {
+				 HtmlParameter cookie = iter.next();
+				 if (cookie.getName().equals(targetToken.getName())) {
+					 cookies.remove(cookie);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: It should use `iter.remove();` instead.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 24]
```diff
@@ -47,6 +51,17 @@ private HttpSender getHttpSender() {
...
+		 if (targetToken.getType() == HtmlParameter.Type.cookie) {
+			 Iterator<HtmlParameter> iter = cookies.iterator();
+			 while (iter.hasNext()) {
+				 HtmlParameter cookie = iter.next();
+				 if (cookie.getName().equals(targetToken.getName())) {
+					 cookies.remove(cookie);
+					 break;
+				 }
+			 }
+		 }
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: In all of the lines above there's an extra whitespace that should be removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 25]
```diff
@@ -62,7 +77,7 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				msg.getRequestHeader().setCookieParams(cookies);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Re-setting all the cookies prevents the regeneration of session (form) tokens (i.e. the same session is used in all requests and the form token ends up being the same).

This changes (both re-setting the cookies as well removing them) could be done, just once, before the for loop, we just need to clone the request, modify it and then use that message (i.e. assign the modified message to instance variable httpMessage).
What do you think?

**Comment by thc202** [comment]
The file ZapAddOn.xml file needs to be updated to reflect the changes (i.e. version bumped, old changes removed and new ones added). [1]

I agree that adding the delay is useful, but its value should be configurable by the user.

[1] https://github.com/zaproxy/zap-extensions/blob/d70ecc6c261f090a9b6a7fbc9f141e22c7d8b29c/src/org/zaproxy/zap/extension/tokengen/ZapAddOn.xml

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 19 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: True, however I prefer to use this approach in case a rewriting is needed and the AbstractTableModel is changed to something else.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 18 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: will be changed in next pull request.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 16 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: Will keep that in mind for next pull requests.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 21 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 22 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Modified.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 23 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Done.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 24 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Nice spotting, removed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 17 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: setCellSelectionEnabled does allow you to copy individual cells, however it does break the highlight as it goes from a row to that particular cell. 
In the end I've decided to remove this function completely as copying the whole row does contain the value. You only have to 'process' it a bit.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java [hunk 26]
```diff
@@ -117,14 +117,14 @@ public Object getValueAt(int rowIndex, int columnIndex) {
...
-        return super.isCellEditable(rowIndex, columnIndex);
-    }
+    // @Override
+    // public boolean isCellEditable(int rowIndex, int columnIndex) {
+    //     if(getColumnClass(columnIndex) == String.class) {
+    //         return true;
+    //     }
+
+    //     return super.isCellEditable(rowIndex, columnIndex);
+    // }
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: Should actually be removed if no longer needed.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java (hunk 16 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenMessagesTableModel.java]: Does that mean that you will leave the indentation changes? Would you mind reverting the changes at least in this file? (which no longer has any code changes)

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 27]
```diff
@@ -47,6 +50,19 @@ private HttpSender getHttpSender() {
...
+			while (iter.hasNext()) {
+				HtmlParameter cookie = iter.next();
+				if (cookie.getName().equals(targetToken.getName())) {
+					iter.remove();
+					break;
+				}
+			}
+		}
+		HttpMessage msg = this.httpMessage.cloneRequest();
+		msg.getRequestHeader().setCookieParams(cookies);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: The cookies should be removed if the token being regenerated is not a cookie (per comment in https://github.com/zaproxy/zap-extensions/pull/295#discussion_r59558486).

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java [hunk 28]
```diff
@@ -58,16 +74,14 @@ protected Void doInBackground() throws Exception {
 			if (this.stopGenerating) {
 				break;
 			}
-
-			HttpMessage msg = this.httpMessage.cloneRequest();
-
+			HttpMessage msgClone = msg.cloneRequest();
+			Thread.sleep(51);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: The value should be configurable.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 27 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Only the cookie that you want to sequence should be removed. It might happen that you want to sequence something after you authenticated, then you want that authentication cookie to be present, i.e. a CSRF token after a login.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 28 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Would you agree that using the "delay when fuzzing" variable is the way to go forward with this, instead of adding a separate value?

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 27 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: I understand that use case, but not removing the cookies will break other use cases. For example, it would no longer be possible to check a form field (e.g. `authenticity_token`) that is bound to the session.
Probably better to allow the user to choose when the cookies are removed or not.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 28 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: We should create another option to not create a dependency between the two add-ons.

**Comment by thc202** [comment]
@vdbaan replied to your comments, sorry for the delay.

**Comment by kingthorin** [comment]
@vdbaan when you address the last of the comments could you also do an Interactive Rebase and fixup into a single commit?

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 27 above)
Comment (vdbaan) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: I will add a selection of this, however I would request of you to create a test case for the scenario that you are describing.

**Comment by vdbaan** [comment]
@kingthorin will do.

File: src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java (hunk 27 above)
Comment (thc202) [inline src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java]: Sure, as unit/integration test for token gen?

**Comment by kingthorin** [comment]
@vdbaan any news?

**Comment by kingthorin** [comment]
@vdbaan any news?

**Comment by vdbaan** [comment]
My apologies for this long delay. have been busy with too many other things.
I'm trying to get the configurable variables in the options menu, however that is more 'interesting' than i initially thought.
Will try better to keep you updated.

**Comment by kingthorin** [comment]
No worries, we get how life gets busy. Let us know if we can help somehow.

**Comment by thc202** [comment]
@vdbaan anything we can do to help with that?

**Comment by kingthorin** [comment]
If it makes sense you could set this PR (branch) to allow modification by maintainers?
https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by thc202** [comment]
I'll resolve the conflicts and try address the review comments.

**Comment by kingthorin** [comment]
I got this all squashed but for some reason when I try to resolve the conflicts and then push it just hangs at 83%

**Comment by thc202** [comment]
Does that still happen?

**Comment by kingthorin** [comment]
Will try again when I get home tonight or in the morning tomorrow and let you know.

**Comment by thc202** [comment]
OK.

**Comment by kingthorin** [comment]
I broke this one while trying to deal with the outstanding items. See https://github.com/zaproxy/zap-extensions/pull/1147 for the new iteration.
